cmd/agent/agent.go

```diff
@@ -0,0 +1,126 @@
+package main
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+
+	// Adds version information
+	_ "github.com/grafana/agent/pkg/build"
+
+	"github.com/grafana/agent/pkg/integrations"
+	"github.com/grafana/agent/pkg/loki"
+	"github.com/grafana/agent/pkg/tempo"
+
+	"github.com/go-kit/kit/log/level"
+	"github.com/grafana/agent/pkg/config"
+	"github.com/grafana/agent/pkg/prom"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/weaveworks/common/server"
+
+	// Register Prometheus SD components
+	_ "github.com/prometheus/prometheus/discovery/install"
+
+	// Register integrations
+	_ "github.com/grafana/agent/pkg/integrations/install"
+
+	"github.com/go-kit/kit/log"
+)
+
+type AgentServer struct {
+	promMetrics *prom.Agent
+	lokiLogs    *loki.Loki
+	tempoTraces *tempo.Tempo
+	manager     *integrations.Manager
+	srv         *server.Server
+}
+
+func NewAgentServer(logger log.Logger, cfg *config.Config) *AgentServer {
```

Comment:
Since we're pulling this out to a separate function, we should return an error here instead of doing `os.Exit(1)` when something fails. It's a good opportunity to also push the error logs back to the caller so this code can be a little cleaner.

i.e.,

```go
srv, err := server.New(cfg.Server) 
if err != nil {
  return nil, fmt.Errorf("failed to create server: %w", err)
}
```